libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthKeychainCredentials.m

```diff
@@ -128,12 +128,6 @@ - (void)setRefreshToken:(NSString *)token withEncryptionKey:(NSData *)encryption
         if (self.isEncrypted) {
             tokenData = [SFSDKEncryptor encryptData:tokenData key:encryptionKey error:nil];
         }
-    } else {
```

Comment:
All the callers of setRefreshToken that sets the refresh token to nil also take care of clearing the other fields. Having such side effect in a setter is not good practice.